Add property type constraints (#597)
neo4j/docs-cypher: modules/ROOT/pages/keyword-glossary.adoc

| Schema
| Create a constraint that ensures all relationships with a particular type have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.

> | xref::constraints/syntax.adoc#constraints-syntax-create-node-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR (n:Label) REQUIRE n.property IS :: TYPE [OPTIONS {}\]]

Reviewer: Maybe clarify that "TYPE" is user input and not just part of the command?
```suggestion
| xref::constraints/syntax.adoc#constraints-syntax-create-node-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR (n:Label) REQUIRE n.property IS :: <TYPE> [OPTIONS {}\]]
```
```suggestion
| xref::constraints/syntax.adoc#constraints-syntax-create-node-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR (n:Label) REQUIRE n.property IS :: type [OPTIONS {}\]]
```

Author: I don't think we have <> anywhere else here so it feels odd to add it. and the reason why I had it in capital letters was because it still keywords that goes in there, but we could have the lowercase letters

Reviewer: maybe "{INT | ...}"?

Author: maybe, but we don't want to add confusion with the {} in the empty options map either (I guess the relationship pattern either looks stupid or just accepts the double use of [])

> | xref::constraints/syntax.adoc#constraints-syntax-create-node-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR (n:Label) REQUIRE n.property IS :: TYPE [OPTIONS {}\]]

Reviewer: In a similar vein as above, I think we should be clearer with the user defined variables being distinct from the rest of the command.
```suggestion
| xref::constraints/syntax.adoc#constraints-syntax-create-node-prop-type[CREATE CONSTRAINT [constraint_name\] [IF NOT EXISTS\] FOR (n:Label) REQUIRE n.foo IS :: TYPE [OPTIONS {}\]]
```
This goes for all constraints, not just the new ones

Author: I'm fine with changing the constraint names but I think we should leave the property and not have `foo` :shrug:, but let's see what the docs team thinks of it, the current names I believe matches the constraint type

| Schema

> | Create a constraint that ensures that for all nodes with a particular label and property, the property have a certain property type (given by `TYPE`).

Author: ```suggestion
| Create a constraint that ensures that for all nodes with a particular label and property, the property have a certain property type (given by `type`).
```

> | Create a constraint that ensures that for all nodes with a particular label and property, the property have a certain property type (given by `type`).

Reviewer 2: ```suggestion
| Create a constraint ensuring that a property belonging to certain nodes has the required property type (given by `type`).
```

Author: This is also following existing patterns (just adding the type requirement) so if we want to update this we should consider if we want to update the other constraints as well

Author: ✅

> | xref::constraints/syntax.adoc#constraints-syntax-create-rel-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR ()-"["r:REL_TYPE"\]"-() REQUIRE r.property IS :: TYPE [OPTIONS {}\]]
> | Schema
> | Create a constraint that ensures that for all relationships with a particular type and property, the property have a certain property type (given by `TYPE`).

Author: ```suggestion
| xref::constraints/syntax.adoc#constraints-syntax-create-rel-prop-type[CREATE CONSTRAINT [property_type\] [IF NOT EXISTS\] FOR ()-"["r:REL_TYPE"\]"-() REQUIRE r.property IS :: type [OPTIONS {}\]]
| Schema
| Create a constraint that ensures that for all relationships with a particular type and property, the property have a certain property type (given by `type`).
```

> | Create a constraint that ensures that for all relationships with a particular type and property, the property have a certain property type (given by `type`).

Reviewer 2: ```suggestion
| Create a constraint ensuring that a property belonging to certain relationship types has the required property type (given by `type`).
```

Author: ✅
